addons/account/models/partner.py

```diff
@@ -821,23 +821,38 @@ def _unlink_if_partner_in_account_move(self):
         if moves:
             raise UserError(_("The partner cannot be deleted because it is used in Accounting"))
 
-    def _increase_rank(self, field, n=1):
-        if self.ids and field in ['customer_rank', 'supplier_rank']:
+    def _increase_rank(self, field: str, n: int = 1):
+        assert field in ('customer_rank', 'supplier_rank')
+        postcommit = self.env.cr.postcommit
+        data = postcommit.data.setdefault(f'account.res.partner.increase_rank.{field}', defaultdict(int))
+        for record in self.sudo():
+            # In case we alrady have a value, we will increase the rank in
+            # postcommit to avoid serialization errors.  However, if the record
+            # has a rank of 0, we increase it directly so that filtering on
+            # whether a partner is a customer or supplier works directly.
```

Comment:
I suppose that we just accept that it could lead to errors sometimes then?
But that it is less likely?

If we accept that, can't we simply use `try_lock_for_update(allow_referencing=True)` then?

Reply:
Ok, so if we can accept this, I will simplify the code and fix the tests that will break.

Reply:
I have a failing test in l10n_nl_reports. We accepted some errors, but now that it is always in post-commit, if we generate files, they may not be correct.
So I think that we need to increase from 0 to 1 in the current transaction. Or am I gettting it wrong?